vendor/brave-ios/Ledger/Models/BATPublisherInfo+Private.mm

```diff
@@ -8,22 +8,43 @@ @implementation BATPublisherInfo (PrivateCpp)
 
 - (ledger::PublisherInfoPtr)cppObj
 {
-  ledger::PublisherInfo info;
-  info.id = std::string(self.id.UTF8String);
-  info.duration = self.duration;
-  info.score = self.score;
-  info.visits = self.visits;
-  info.percent = self.percent;
-  info.weight = self.weight;
-  info.excluded = (ledger::PUBLISHER_EXCLUDE)self.excluded;
-  info.category = (ledger::REWARDS_CATEGORY)self.category;
-  info.reconcile_stamp = self.reconcileStamp;
-  info.verified = self.verified;
-  info.name = std::string(self.name.UTF8String);
-  info.url = std::string(self.url.UTF8String);
-  info.provider = std::string(self.provider.UTF8String);
-  info.favicon_url = std::string(self.faviconUrl.UTF8String);
-  return info.Clone();
+  auto info = ledger::PublisherInfo::New();
+  info->id = std::string(self.id.UTF8String);
+  info->duration = self.duration;
+  info->score = self.score;
+  info->visits = self.visits;
+  info->percent = self.percent;
+  info->weight = self.weight;
+  info->excluded = (ledger::PUBLISHER_EXCLUDE)self.excluded;
+  info->category = (ledger::REWARDS_CATEGORY)self.category;
+  info->reconcile_stamp = self.reconcileStamp;
+  info->verified = self.verified;
+  info->name = std::string(self.name.UTF8String);
```

Comment:
Just a preference thing I guess.. but std::string constructor doesn't need to be called :)
When you do an assignment with a const char* to an std::string, it copies it automatically.

IE:  `info->name = self.name.UTF8String` vs. `info->name = std::string(self.name.UTF8String)`

The first one will assign directly to name. The second one will create an std::string from self.name and call the move constructor to move it into info->name.

Both equivalent though.

Reply:
Oh that's awesome, I didn't realize that! I assume this only works when the left hand side is specifically an std::string and not being inferred by auto, right?

Reply:
> Oh that's awesome, I didn't realize that! I assume this only works when the left hand side is specifically an std::string and not being inferred by auto, right?

Correct. If it's auto then it'll just copy the "pointer" and not its contents. If it's specifically `std::string` on the left hand side, it'll copy the contents of the pointer into the left hand `std::string` directly :)

Another one is for example:

```c++
std::string foo;

auto &f = foo;
f = "hello";  //copies directly. No need because even though we used auto, it's still referring to an std::string.
```

As long as left side is an std::string or refers to one, then it copies directly without the move constructor :D